build-llvm.sh

```diff
@@ -68,15 +68,22 @@ if [ -z "$CHECKOUT_ONLY" ]; then
 fi
 
 if [ ! -d llvm-project ]; then
-    # When cloning master and checking out a pinned old hash, we can't use --depth=1.
-    git clone https://github.com/llvm/llvm-project.git
+    mkdir llvm-project
+    cd llvm-project
+    git init
+    git remote add origin https://github.com/llvm/llvm-project.git
+    git fetch --depth 1 origin "$LLVM_VERSION"
+    git checkout FETCH_HEAD
+    cd ..
     CHECKOUT=1
```

Comment:
We probably should do `unset SYNC` here now too, to make sure we don't do the shallow fetch twice if running with `SYNC=1` set for the initial clone/fetch, as doing a shallow fetch of the same reference twice in a row ends up downloading it twice.